### PR TITLE
Vulkan-hpp: instance, extensions & validation layers are refactored to vulkan-hpp

### PR DESCRIPTION
**Pull request by MoonDweller29**

### CODE REVIEW

File: include/magma/app/App.h [hunk 1]
```diff
@@ -36,7 +36,7 @@ class App {
     std::vector<Vertex> vertices;
     std::vector<uint32_t> indices;
 
-    std::unique_ptr<VkInstanceHolder>     instance;
+    std::unique_ptr<Context>     instance;
     std::unique_ptr<DebugMessenger>       debugMessenger;
```
Comment (Nastormo) [inline include/magma/app/App.h:40]: May use a similar alignment

File: include/magma/vk/Context.h [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#pragma once
+#include <vulkan/vulkan.hpp>
+
+class Context {
+public:
+    const vk::Instance &instance(){ return _instance; }
+    const VkInstance &c_instance(){ return _c_instance; } //only for Vulkan C API
```
Comment (Nastormo) [inline include/magma/vk/Context.h:7]: May add space? &instance() {}

File: include/magma/vk/vulkan_common.h [hunk 3]
```diff
@@ -1,5 +1,5 @@
 #pragma once
-#include <vulkan/vulkan.h>
+#include <vulkan/vulkan.hpp>
 #include <stdexcept>
 #include <vector>
 #include "magma/app/log.hpp"
```
Comment (Nastormo) [inline include/magma/vk/vulkan_common.h:5]: may add \n between standard lib and custom lib

**Review by Nastormo — CHANGES_REQUESTED** [review]

File: include/magma/vk/vulkan_common.h (hunk 3 above)
Comment (MoonDweller29) [reply include/magma/vk/vulkan_common.h:5]: no actual need for this. There only one include from our lib

File: include/magma/app/App.h (hunk 1 above)
Comment (MoonDweller29) [reply include/magma/app/App.h]: done

File: include/magma/vk/Context.h (hunk 2 above)
Comment (MoonDweller29) [reply include/magma/vk/Context.h]: done

File: include/magma/vk/Context.h [hunk 4]
```diff
@@ -0,0 +1,14 @@
+#pragma once
+#include <vulkan/vulkan.hpp>
+
+class Context {
+public:
+    const vk::Instance &instance()   { return _instance; }
+    const VkInstance   &c_instance() { return _c_instance; } //only for Vulkan C API
```
Comment (Nastormo) [inline include/magma/vk/Context.h:7]: May use https://en.cppreference.com/w/cpp/language/attributes/deprecated

File: include/magma/vk/Context.h [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+    Context();
+    ~Context();
```
Comment (Nastormo) [inline include/magma/vk/Context.h:10]: Usually this method first

File: include/magma/vk/vulkan_common.h (hunk 3 above)
Comment (Nastormo) [reply include/magma/vk/vulkan_common.h:5]: It creates some order.
Usually

1. Custom h file which contain class definition 
2. Standard lib 
3. Custom lib

File: include/magma/vk/validationLayers/DebugMessenger.h [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+class DebugMessenger {
+public:
+    static Log::Level msgSeverityToLogLevel(vk::DebugUtilsMessageSeverityFlagBitsEXT msgSeverity);
+    static vk::DebugUtilsMessengerCreateInfoEXT getCreateInfo();
+
+    DebugMessenger(const vk::Instance &instance);
+    ~DebugMessenger();
+private:
+    vk::DebugUtilsMessengerEXT debugMessenger;
```
Comment (Nastormo) [inline include/magma/vk/validationLayers/DebugMessenger.h:13]: Need replace _debugMessenger

**Review by Nastormo — CHANGES_REQUESTED** [review]

File: include/magma/vk/validationLayers/DebugMessenger.h (hunk 6 above)
Comment (MoonDweller29) [reply include/magma/vk/validationLayers/DebugMessenger.h]: done

**Review by Nastormo — APPROVED** [review]

**Pull request merged**